src/mixer/audio.ts

```diff
@@ -156,15 +166,28 @@ class Player extends ((PlayerEmitter as unknown) as { new (): EventEmitter }) {
   _applyVolume() {
     const level = this.volume + this.trim;
     const linear = Math.pow(10, level / 20);
-    if (linear < 1) {
-      this.wavesurfer.setVolume(linear);
-      if (!this.customOutput) {
-        (this.wavesurfer as any).backend.gainNode.gain.value = 1;
-      }
+
+    // Actually adjust the wavesurfer gain node gain instead, so we can tap off analyser for PFL.
+    this.wavesurfer.setVolume(1);
+    if (!this.customOutput) {
+      (this.wavesurfer as any).backend.gainNode.gain.value = linear;
+    }
+  }
+
+  _connectPFL() {
+    if (this.pfl) {
+      // In this case, we just want to route the player output to the headphones direct.
+      // Tap it from analyser to avoid the player volume.
+      (this.wavesurfer as any).backend.analyser.connect(
+        this.engine.headphonesNode
+      );
     } else {
-      this.wavesurfer.setVolume(1);
-      if (!this.customOutput) {
-        (this.wavesurfer as any).backend.gainNode.gain.value = linear;
+      try {
+        (this.wavesurfer as any).backend.analyser.disconnect(
+          this.engine.headphonesNode
+        );
+      } catch (e) {
+        // This connection wasn't connected anyway, ignore.
```

Comment:
Worth at least a `console.warn()`?

Reply:
Hmm. I could do, but it is kinda expected behaviour due to the way it's written.